src/field_ground_effect.c

```diff
@@ -1,11 +1,126 @@
+    MetatileBehavior_IsPuddle,
+    MetatileBehavior_IsSurfableWaterOrUnderwater,
+    MetatileBehavior_IsShallowFlowingWater,
+    MetatileBehavior_IsATile,
+};
+
+static const u32 jumpLandingFlags[] = {
+    0x00001000, // Landing in tall grass
+    0x00002000, // Landing in long grass
+    0x00004000, // Landing on puddle
+    0x00008000, // Landing on surfable water or underwater
+    0x00004000, // Landing on shallow flowing water
+    0x00010000, // Landing on any other type of ground
+};
+
+static bool8 (*const gUnknown_08376040[])(u8) = {
+    MetatileBehavior_IsJumpSouth,
+    MetatileBehavior_IsJumpNorth,
+    MetatileBehavior_IsJumpWest,
+    MetatileBehavior_IsJumpEast,
+};
+
+static const u8 gUnknown_08376050[] = {
+    0x73, 0x73, 0x53, 0x73, 0x53, 0x73, 0x53, 0x73, 0x53, 0x73, 0x53, 0x73, 0x53, 0x00, 0x00, 0x73
+};
+
+// Each byte corresponds to a sprite priority for a field object.
+// This is directly the inverse of gFieldObjectPriorities_08376070.
+static const u8 gFieldObjectPriorities_08376060[] = {
+    2, 2, 2, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 0, 0, 2
+};
+
+// Each byte corresponds to a sprite priority for a field object.
+// This is the inverse of gFieldObjectPriorities_08376060.
+// 1 = Above player sprite
+// 2 = Below player sprite
+static const u8 gFieldObjectPriorities_08376070[] = {
+    1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 0, 0, 1,
+};
+
+static void (*const gUnknown_08376080[])(struct MapObject *mapObj, struct Sprite *sprite, u8 a) = {
+    nullsub, DoTracksGroundEffect_Footprints, DoTracksGroundEffect_BikeTireTracks,
+};
+
+// First byte is a Field Effect script id. (gFieldEffectScriptPointers)
+// Last three bytes are unknown.
+static const u8 gSandFootprints_FieldEffectData[] = { 0xD, 0x0, 0x18, 0x0 };
+
+//  Specifies which bike track shape to show next.
+//  For example, when the bike turns from up to right, it will show
+//  a track that curves to the right.
+//  Each 4-byte row corresponds to the initial direction of the bike, and
+//  each byte in that row is for the next direction of the bike in the order
+//  of down, up, left, right.
+static const u8 gBikeTireTracks_Transitions[4][4] = {
+    1, 2, 7, 8, 1, 2, 6, 5, 5, 8, 3, 4, 6, 7, 3, 4,
+};
+
+static void (*const gUnknown_083760A0[])(
+    struct MapObject *mapObj, struct Sprite *sprite) = { GroundEffect_SpawnOnTallGrass,
+    sub_8063E94,
+    sub_8063EE0,
+    sub_8063F2C,
+    GroundEffect_WaterReflection,
+    GroundEffect_IceReflection,
+    GroundEffect_FlowingWater,
+    sub_8063FA0,
+    sub_8063FCC,
+    GroundEffect_Ripple,
+    GroundEffect_StepOnPuddle,
+    GroundEffect_SandPile,
+    GroundEffect_JumpOnTallGrass,
+    GroundEffect_JumpOnLongGrass,
+    GroundEffect_JumpOnShallowWater,
+    GroundEffect_JumpOnWater,
+    GroundEffect_JumpLandingDust,
+    GroundEffect_ShortGrass,
+    GroundEffect_HotSprings,
+    GroundEffect_Seaweed };
+
+#define NUM_GROUND_EFFECTS (sizeof(gUnknown_083760A0) / sizeof(gUnknown_083760A0[0]))
```

Comment:
We have an ARRAY_COUNT macro which does this, so you can just do `ARRAY_COUNT(gUnknown_083760A0)`.